### DISCUSSION TITLE
Why mutex is used in ServerCalls?

### DISCUSSION THREAD

**Discussion by bagjinok** [grpc/grpc-kotlin#279] [Q&A]
Hi 😃  grpc-kotlin team! 
I'm a very newbie on Grpc and Kotlin!
I just wondering to know that why mutex is used in `ServerCalls`.
If it’s not too much to ask, could you please explain why mutex is used?
https://github.com/grpc/grpc-kotlin/blob/master/stub/src/main/java/io/grpc/kotlin/ServerCalls.kt#L243

**Comment by lowasser** [comment]
If you look in the revision history, you can see the commit linked to the bug that describes a race that could have happened otherwise.  https://github.com/grpc/grpc-kotlin/commit/ef06b07a84256a116b16afa2fd3d7aecf4aa402f#diff-8f7b388f007f7822b342a47efedb08d735860ccdc4aeaec73a0d09406d96f54f
